### PR TITLE
[AL-7738] Support custom metrics for predictions

### PR DESCRIPTION
**Pull request by kozikkamil**
Adds "customMetrics" field definition for ndjson and "custom_metrics" for class object for annotations.
It's meant to be used at the prediction level in when adding predictions to MEA.

Should not be merged without respective changes deployed to rest of tech stack.

### CODE REVIEW

File: labelbox/data/mixins.py [hunk 1]
```diff
@@ -1,32 +1,70 @@
...
+                "Confidence is not supported for this annotation type yet"
+            )
+        return super().__new__(cls)
+
+
+class CustomMetric(BaseModel):
+    name: str
+    value: float
+
+    @validator("name")
```
Comment (vbrodsky) [inline labelbox/data/mixins.py:39]: yep this is a good catch... pydantic will coerce to str any other possible type... I know in pydantic2 there is a config to make it more strict.  I did not see anything like this in pydantic 1. Good catch

File: labelbox/data/mixins.py [hunk 2]
```diff
@@ -1,32 +1,70 @@
...
+        return value
+
+
+class CustomMetricsMixin(BaseModel):
+    custom_metrics: Optional[List[CustomMetric]] = None
+
+    def dict(self, *args, **kwargs):
+        res = super().dict(*args, **kwargs)
+
+        if "customMetrics" in res and res["customMetrics"] is None:
```
Comment (vbrodsky) [inline labelbox/data/mixins.py:58]: so, to be clear, we do not want to serlaize custom metrics?

File: tests/integration/annotation_import/test_mea_prediction_import.py [hunk 3]
```diff
@@ -185,6 +186,29 @@ def test_create_from_local_file(tmp_path, model_run_with_data_rows,
     assert annotation_import.state == AnnotationImportState.FINISHED
     annotation_import_test_helpers.download_and_assert_status(
         annotation_import.status_file_url)
+    
+
+def test_predictions_with_custom_metrics(model_run,
```
Comment (vbrodsky) [inline tests/integration/annotation_import/test_mea_prediction_import.py:191]: is this an additional test or is it something specific to the metrics use case?

**Review by vbrodsky — COMMENTED** [review]
Solid work, can approve when the time comes, left a couple of question for my own curiosity

File: labelbox/data/mixins.py (hunk 2 above)
Comment (kozikkamil) [reply labelbox/data/mixins.py:58]: Could you elaborate a bit please? This particular line is just making sure we don't pass None, but rather remove the key entirely in that case

File: tests/integration/annotation_import/test_mea_prediction_import.py (hunk 3 above)
Comment (kozikkamil) [reply tests/integration/annotation_import/test_mea_prediction_import.py:191]: It checks whether a mea import with custom metrics on prediction level succeeds

File: labelbox/data/mixins.py (hunk 2 above)
Comment (vbrodsky) [reply labelbox/data/mixins.py:58]: Sorry IS NONE... missed it

**Review by vbrodsky — APPROVED** [review]

**Pull request merged**